src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -893,6 +912,27 @@ public <T> T[] toArray(T[] a) {
             }
             return array;
         }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void forEach(Consumer<? super E> action) {
+            Objects.requireNonNull(action);
+            if (REVERSE && e1 != EMPTY) {
+                action.accept((E) e1);
+            }
+            action.accept(e0);
+            if (!REVERSE && e1 != EMPTY) {
+                action.accept((E) e1);
+            }
```

Comment:
I'm curious to know how the following alternative would fare:

```suggestion
            if (e1 != EMPTY) {
                action.accept(REVERSE ? (E)e1 : (E)e0); // implicit null check
                action.accept(REVERSE ? (E)e0 : (E)e1);
            } else {
                action.accept(e0); // Implicit null check
            }
```

Reply:
I used this:
```java
            if (e1 == EMPTY) {
                action.accept(e0); // Implicit null check
            } else {
                action.accept(REVERSE ? (E)e1 : e0); // implicit null check
                action.accept(REVERSE ? e0 : (E)e1);
            }
```

The results are about the same as the current patch.
With the alternative:
```
Benchmark                        Mode  Cnt    Score    Error   Units
ImmutableColls.forEachOverList  thrpt   15  367.651 ± 10.730  ops/us
ImmutableColls.forEachOverSet   thrpt   15   84.397 ± 12.235  ops/us
ImmutableColls.getOrDefault     thrpt   15  243.874 ±  1.891  ops/us
ImmutableColls.iterateOverList  thrpt   15  149.525 ±  3.652  ops/us
ImmutableColls.iterateOverSet   thrpt   15   60.202 ±  5.053  ops/us
```
Current patch:
```
Benchmark                        Mode  Cnt    Score   Error   Units
ImmutableColls.forEachOverList  thrpt   15  364.515 ± 8.076  ops/us
ImmutableColls.forEachOverSet   thrpt   15   86.012 ± 2.420  ops/us
ImmutableColls.getOrDefault     thrpt   15  243.723 ± 1.071  ops/us
ImmutableColls.iterateOverList  thrpt   15  149.392 ± 5.328  ops/us
ImmutableColls.iterateOverSet   thrpt   15   63.579 ± 4.671  ops/us
```

Reply:
@viktorklang-ora I've updated this piece of code, does it look better now?